dev/AppNotifications/AppNotificationBuilder/AppNotificationButton.cpp

```diff
@@ -11,8 +11,8 @@ namespace winrt::Microsoft::Windows::AppNotifications::Builder::implementation
 
     winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationButton AppNotificationButton::AddArgument(winrt::hstring const& key, winrt::hstring const& value)
     {
-        THROW_HR_IF(E_INVALIDARG, key.empty());
-        THROW_HR_IF(E_INVALIDARG, m_protocolUri);
+        THROW_HR_IF_MSG(E_INVALIDARG, key.empty(), "You must provide a key when adding an argument.");
+        THROW_HR_IF_MSG(E_INVALIDARG, m_protocolUri, "You cannot add an argument after calling SetInvokeUri.");
```

Comment:
NIT: `!m_protocolUri.empty()` ?

Reply:
So in AppNotification xml, you cannot set both Arguments and a ProtocolUri for activation. This is basically saying you can only have one or the other. The reverse check appears in SetInvokeUri as well which verifies AddArguments hasn't been called.

Reply:
Since this is a winrt::Uri isn't it ok to do it this way? Otherwise, it will have to be m_protocolUri.ToString().empty() or something like it